### ISSUE TITLE
Possible Enhancements

### ISSUE THREAD

**Issue by SukkoPera** [SukkoPera/Raemixx500#14]
The first goal of this project is to produce a board equivalent to that of a standard A500+. This will allow to replace the board of a failing machine by salvaging components from it. It will also validate the schematics and general design.

Some deviations from the original design are expected (see #13), but basically it will be the same board.

After the first prototypes are confirmed working we can make changes to the board to bring the A500 design to the 21st century. This ticket shall collect ideas for upgrades.

1. 2 MB Chip RAM onboard. This is simple, adding some RAM chips connected to /BRAS1 should be enough. It can either be done with the usual SOJ-40 chips or with a single 2 MB SOJ-42 chip. The latter would save quite a bit of board space.

2. Since the above would render the trapdoor slot useless, the trapdoor connector could be rewired so that it allows plugging some Slow RAM, as on earlier mainboard revisions. This can be discussed with PeteAU and the other guys who made the Open Design expansion on EAB.
EDIT: Someone mentioned the trapdoor connector could still be useful for an expansion board like the A314. But would it still work if we had 2 MB Chip RAM on the mainboard?

3. 8 MB Fast RAM onboard. This can be done using [lvd's design](http://lvd.nedopc.com/Projects/a600_8mb/index.html) or together with IDE and acceleration as detailed below.

4. IDE. Shall we replace Gary with Gayle or... (see below)?

5. Acceleration. We could get this and the above (and Fast RAM) with help from Mr Leary and his TerribleFires. Not sure it is easily doable on a 2-layer board though.

~6. Use drive/power led for HD activity. See [here](https://lyonsden.net/amiga-500-ide-activity-led/).~ Done in V2.

~7. Integrate a [Kickstart Switcher](https://github.com/SukkoPera/OpenKickstartSwitcher).~ Done in ab97624eb4c146b23ac98462a14d15b203cdb8f8.

~8. Integrate a [Drive Switcher](https://github.com/SukkoPera/OpenAmigaDriveSwitcher).~ Done in c3e65738cd11ce028346485becdd2adaefab6da4.

9. Integrate a Gotek Drive ([Schematics here](https://user-images.githubusercontent.com/1316870/56572513-7f468e00-65bf-11e9-8cd3-df50e27f0909.png), firmware would be [FlashFloppy](https://github.com/keirf/FlashFloppy) of course).

10. Color composite output. Not sure we should do this onboard, as I would like to do it in [OpenAmigaVideoHybrid](https://github.com/SukkoPera/OpenAmigaVideoHybrid) through an [AD724 chip](https://www.analog.com/media/en/technical-documentation/data-sheets/ad724.pdf), so that *any* A500 could get this nice addition, as well as a possible S-Video output.

11. Replace (or better integrate) the joystick/mouse ports with USB ports. This can be done with a single STM32F105 chip, just like [HID2AMI](https://github.com/EmberHeavyIndustries/HID2AMI), but we would need to write an GPL'd firmware for that first. Doable but a long job on its own.

12. Clock port. What could this be useful for? USB?

13. Integrate a PlipBox Ethernet card.

If we need to gain space on the board, we can replace the CPU and/or ROM with PLCC versions or use SOIP chips, but this would require quite a bit of rewiring.

Another idea could be making an ATX-format Amiga board, maybe with some Zorro II expansions slot as well.

**Comment by SukkoPera** [comment]
7 and 8 are done.

**Comment by stas2k** [comment]
I like points 3 and 4 together. It makes an A500 a good ClassicWB machine. Was even looking into combining these 2 into a single board with a shared CPLD. What broke me was the ide68k source written in ABEL :)

Integrating Gotek is also a good ideas, as you can wire all the OSD stuff on the motherboard itself.

**Comment by SukkoPera** [comment]
I'm not going to do any new work on this (or any other, actually) project. If anyone wants to have a go at any of these points, feel free to do it.

**Comment by stas2k** [comment]
Understood. Thanks for your work in this project. It might help a lot of people with dead A500+. I managed to revive mine with some wires, otherwise would already be sending you patches ;)

**Comment by weirdgyn** [comment]
Why not put the option for a 68010 processor (I think is pretty the same maybe only an higher clock rate) ?
I remember some of my high-school friends using it ....

**Comment by SukkoPera** [comment]
I don't think the 68010 requires any particular hardware support, it should be a drop-in replacement for the 68000. Anyone who builds this can already use it straight away.

**Comment by weirdgyn** [comment]
Yep it should be pin-2-pin compatibile that's sure but if I remember well my friends used to clock it much higher than ordinary A500 68k (7MHz?) they where probably using a clock doubler since 68010 was already running up to 16MHz.

**Comment by SukkoPera** [comment]
Well, I have no idea of how to do that. If you want to contribute, you are welcome to join the project.

**Comment by kipper2k** [comment]
I would like to do a 2MB chipmem for this board, i have lots of the 2MB SOJ 42 chips left over from my 8MB fastmem/IDE project, does anyone have a schematic on how to plumb this in, so i can give it a shot. Thanx

**Comment by ZaPpi3** [comment]
The SMD500+ project implements this.
https://github.com/gazzmaniac/Amiga-SMD-500


________________________________

**Comment by SukkoPera** [comment]
I have a piggyback board that I used for some experiments. It has a switch to select 1 or 2 MB of chip RAM with a single SOJ-42 chip. Both configurations seem to work great, but for some reason, when it's set to 1 MB, plugging a trapdoor expansion will result in odd video issues.

Once that is debugged, it would be very nice to integrate this feature into Raemixx.

I also have #10 done, and #11 wuld be around 95% done (on a separate board, to be integrated).

Too bad I got sick of it all.

**Comment by Silveriomrs** [comment]
Change some jumpers footprints for SMD Swtichs footprints. That way who want it, could install and SMD switch and vary the setup of the board without ironsolder, that at the end could damage the PCB.

**Comment by SukkoPera** [comment]
The easiest thing would be to turn them into pin headers so that normal jumper caps could be used, still allowing for fancier setups with switches . That said, most of the jumpers will never be touched more than once/twice during the whole lifetime of the board, and the majority will never be touched at all, so I'm not really sure this would be very useful.

Anyway, feel free to go for what you prefer, I'm never ever going to touch anything Amiga again.

**Comment by Lensrev** [comment]
@SukkoPera Thank you for this beautiful board. What happened that made you not want to touch anything Amiga again?

**Comment by SukkoPera** [comment]
Thanks @Lensrev, it's a combination of many different things that convinced me to leave the Amiga scene. Although it's a long discussion that I'd rather have in front of a beer rather than here :).
